### PR TITLE
Support configurable domain cluster name

### PR DESCRIPTION
**Pull request by vanyarock01**
By default domain cluster name is `cluster.local`.
To change add variable `ClusterDomainName` to cartridge `values.yaml`. For example, :
```yaml
ClusterEnv: dev
ClusterName: test-app
ClusterDomainName: domain.name
...
```

Using the minikube for testing: 
```
minikube start --dns-domain='custom.domain' --extra-config='kubelet.cluster-domain=custom.domain'
```

### CODE REVIEW

File: examples/kv/helm-chart/templates/deployment.yaml [hunk 1]
```diff
@@ -53,6 +53,7 @@ spec:
     metadata:
       labels:
         tarantool.io/cluster-id: {{ $.Values.ClusterName }}
+        tarantool.io/cluster-domain-name: "{{ default "cluster.local" $.Values.ClusterDomainName }}"
```
Comment (vasiliy-t) [inline examples/kv/helm-chart/templates/deployment.yaml:56]: default should be defined in values.yaml

File: pkg/topology/builtin.go [hunk 2]
```diff
@@ -191,9 +191,17 @@ func GetRoles(pod *corev1.Pod) ([]string, error) {
 // Join comment
 func (s *BuiltInTopologyService) Join(pod *corev1.Pod) error {
 
-	advURI := fmt.Sprintf("%s.%s.%s.svc.cluster.local:3301", pod.GetObjectMeta().GetName(), s.clusterID, pod.GetObjectMeta().GetNamespace())
-
 	thisPodLabels := pod.GetLabels()
+	clusterDomainName, ok := thisPodLabels["tarantool.io/cluster-domain-name"]
+	if !ok {
+		return errors.New("cluster domain name empty")
```
Comment (vasiliy-t) [inline pkg/topology/builtin.go:197]: this looks like breaking change

**Review by vasiliy-t — CHANGES_REQUESTED** [review]

**Review by vasiliy-t — APPROVED** [review]

**Comment by vanyarock01** [comment]
@timjohncurtis Take a look, please.

**Review by timjohncurtis — APPROVED** [review]
yep looks good to me 👍

**Pull request merged**